src/cli/commands/force/lightning/local/setup.ts

```diff
@@ -81,13 +87,19 @@ export default class Setup extends SfdxCommand {
         await LoggerSetup.initializePluginLoggers();
     }
 
-    protected setup(): BaseSetup {
-        let setup: BaseSetup = ({} as any) as BaseSetup; // should not be the case due to prior validation.
-        if (CommandLineUtils.platformFlagIsAndroid(this.flags.platform)) {
-            setup = new AndroidEnvironmentSetup(this.logger);
-        } else if (CommandLineUtils.platformFlagIsIOS(this.flags.platform)) {
-            setup = new IOSEnvironmentSetup(this.logger);
+    protected addRequirements(reqs: Requirement[]) {
```

Comment:
This adds ability for any class that extends Setup, to be able to append its own requirements on top the base setup requirements